jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -810,6 +813,29 @@ public static String getCanonicalNameForWebAppTmpDir(WebAppContext context)
 
         return StringUtil.sanitizeFileSystemName(canonicalName.toString());
     }
+    
+    private static String getResourceBasePath(Resource resource)
+    {
+        String tmp = "";
+        try
+        {
+            tmp = URIUtil.decodePath(resource.getURI().getPath());
+        }
+        catch (Exception e)
+        {
+            try
+            {
+                tmp = URIUtil.decodePath(resource.getURI().toURL().getPath());
```

Comment:
This is invalid.
When a double decode happens you will have a new directory that is unrelated to the original resource base.

An example:

You have, an on disk name of ...

```
C:\Program Files\jenkins\workspaces\the%2Fbranch\job\43\projectname\
```

Which gets encoded into URI form as ...

```
file:///C:/Program%20Files/jenkins/workspaces/the%252Fbranch/job/43/projectname/
```

Using URI.getPath() results in ...

```
C:/Program%20Files/jenkins/workspaces/the%252Fbranch/job/43/projectname/
```

Using URI.toURL() results in ...

```
file:///C:/Program Files/jenkins/workspaces/the%2Fbranch/job/43/projectname/
```

Using URI.toURL().getPath() results in ...

```
C:/Program Files/jenkins/workspaces/the%2Fbranch/job/43/projectname/
```

Using URIUtil.decodePath() on that results in ...

```
C:/Program Files/jenkins/workspaces/the/branch/job/43/projectname/
```

You've now changed the directory name, and Jenkins can no longer clean up after itself.

Reply:
I don't know what this has to do with Jenkins.  When the webapp is stopped/destroyed the WebInfConfiguration.deconfigure method will delete the tmp directory, regardless of whatever it is called if context.isPersistTempDirectory() is false. Are you saying that Jenkins persists the tmp directory then does some manual deletion of the tmp directory?

Reply:
Jenkins is an extremely common situation of encoding special characters for use on filesystem paths.
It's one of the more common sources for our bugs.

Do this ... create a branch called `jan/feature-foo` (which is valid git, and a syntax encouraged by many organizations).
When a CI kicks in for that branch, the `/` will be encoded as `%2F` on disk.  It will look like `workspaces/jan%2Ffeature-foo/job/1` on disk.
When we take that, and decode it arbitrarily, we get a double-decode, which is invalid.

Incidentally, if you do this, the jetty-documentation will not build as asciidoctor has the same bug we've fixed in various places in Jetty.

Reply:
@joakime I still don't understand what Jenkins has to do with this. In the example you provided, the only part of that name that we would use in creating the name of the temporary directory is "projectname".